concent_api/concent_api/middleware.py

```diff
@@ -39,3 +43,33 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+class HandleServerErrorMiddleware(object):
+    """
+    Used to catch all unhandled exceptions and return JSON response containing error information.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        try:
+            sid = transaction.savepoint()
+            response = self.get_response(request)
+            transaction.savepoint_commit(sid)
+            return response
+        except Exception as exc:  # pylint: disable=broad-except
+            transaction.savepoint_rollback(sid)
+            logging.logger.exception(exc)
+            return self._build_json_response(exc)
+
+    @staticmethod
```

Comment:
Will this ever be used as a static method ?

Reply:
I do not think so... Would you prefer to have me disable pylin warning? ;)

Reply:
I didn't notice that, then we can keep it.

Reply:
ok, so I will revert it to `@staticmethod` :)

Reply:
done